esp-hal-common/src/uart.rs

```diff
@@ -1543,57 +1587,23 @@ mod asynch {
         }
     }
 
+    /// Interrupt handler for all UART instances
+    /// Clears and disables interrupts that have occured and have their enable
+    /// bit set The fact that an interrupt has been disabled is used by the
+    /// futures to detect that they should indeed resolve after being woken up
     fn intr_handler(uart: &RegisterBlock) -> bool {
-        let int_raw_val = uart.int_raw.read();
-        let int_ena_val = uart.int_ena.read();
-
-        let mut wake = false;
+        let occured_interrupts = UartInterrupt::from_bits_truncate(uart.int_raw.read().bits());
```

Comment:
I do like this cleanup, but I think we can do it without `bitfield`, by just doing some manual bit manipulation.